components/raftstore-v2/src/worker/checkpoint.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    fmt::Display,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use engine_traits::{Checkpointer, KvEngine, TabletRegistry};
+use futures::channel::oneshot::Sender;
+use raftstore::store::RAFT_INIT_LOG_INDEX;
+use slog::Logger;
+use tikv_util::{slog_panic, time::Instant, worker::Runnable};
+
+use crate::operation::SPLIT_PREFIX;
+
+pub enum Task {
+    Checkpoint {
+        // it is only used to assert
+        cur_suffix: u64,
+        log_index: u64,
+        parent_region: u64,
+        split_regions: Vec<u64>,
+        sender: Sender<Duration>,
+    },
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Task::Checkpoint {
+                log_index,
+                parent_region,
+                split_regions,
+                ..
+            } => write!(
+                f,
+                "create checkpoint for batch split, parent region_id {}, source region_ids {:?}, log_index {}",
+                parent_region, split_regions, log_index,
+            ),
+        }
+    }
+}
+
+pub struct Runner<EK: KvEngine> {
+    logger: Logger,
+    tablet_registry: TabletRegistry<EK>,
+}
+
+pub fn temp_split_path<EK>(registry: &TabletRegistry<EK>, region_id: u64) -> PathBuf {
+    let tablet_name = registry.tablet_name(SPLIT_PREFIX, region_id, RAFT_INIT_LOG_INDEX);
+    registry.tablet_root().join(tablet_name)
+}
+
+impl<EK: KvEngine> Runner<EK> {
+    pub fn new(logger: Logger, tablet_registry: TabletRegistry<EK>) -> Self {
+        Self {
+            logger,
+            tablet_registry,
+        }
+    }
+
+    fn checkpoint(
+        &self,
+        parent_region: u64,
+        split_regions: Vec<u64>,
+        cur_suffix: u64,
+        log_index: u64,
+        sender: Sender<Duration>,
+    ) {
+        let now = Instant::now();
+
+        let mut cache = self.tablet_registry.get(parent_region).unwrap();
+        let tablet = cache.latest().unwrap();
+        let (_, _, suffix) = self
+            .tablet_registry
+            .parse_tablet_name(Path::new(tablet.path()))
+            .unwrap();
+        assert_eq!(cur_suffix, suffix);
```

Comment:
add the tablet path and region id information into the assert_eq! message, so that we can do further troubleshooting.